### DISCUSSION TITLE
BasicVSR++ as generator in GAN

### DISCUSSION THREAD

**Discussion by NasrinR791** [open-mmlab/mmediting#1201] [How-to]
Hi,

I want to use BasicVSR++ as generator in GAN framework, how can I do this in mmediting?
Also, How can I add customized loss functions in training? Is there any tutorial for this kind of modifications?

I would appreciate if you could help me.

**Comment by zengyh1900** [comment]
Hi @NasrinR791 

Please have a look here: https://mmediting.readthedocs.io/en/latest/tutorials/customize_models.html 
You can customize your model, of course including adding a discriminator to BasicVSR++. 

You can also customize loss functions following this tutorial: https://mmediting.readthedocs.io/en/latest/tutorials/customize_losses.html 

I hope it helps!
